src/zzz_od/application/shiyu_defense/shiyu_defense_app.py

```
@@ -97,35 +97,25 @@ def choose_node_idx(self) -> OperationRoundResult:
+            # OCR 可能将 "2/5" 连写为 "215"，取首尾数字分别作为当前和总节点数。
+            digits = re.findall(r'\d', progress_text)
+            if len(digits) >= 2:
+                current = int(digits[0])
+                total = int(digits[-1])
                 log.info('剧变节点进度 %d/%d', current, total)
                 self.config.critical_max_node_idx = total
                 next_idx = current + 1
                 if next_idx > total:
                     return self.round_success(ShiyuDefenseApp.STATUS_ALL_FINISHED)
+            elif len(digits) == 1:
+                # 只有一个数字，可能是 "5"，说明已完成
```

Ruff (0.15.21)

[warning] Comment contains ambiguous `，` (FULLWIDTH COMMA). Did you mean `,` (COMMA)?

(RUF003)